src/main/java/frc/robot/commands/scoring/intake/IntakeGamePiece.java

```diff
@@ -2,34 +2,48 @@
 
 import edu.wpi.first.wpilibj2.command.CommandBase;
 import frc.robot.constants.IntakeConstants;
-import frc.robot.subsystems.Intake;
+import frc.robot.subsystems.RollerIntake;
+import frc.robot.subsystems.RollerIntake.IntakeMode;
+import frc.robot.subsystems.RollerIntake.IntakePiece;
 
 public class IntakeGamePiece extends CommandBase {
-  Intake m_intake; 
+
+  private final RollerIntake m_intake; 
+  private final IntakePiece m_type;
 
   /**
    * Spins the intake until the game piece is inside the intake.
    * @param intake the intake subsystem
    */
-  public IntakeGamePiece(Intake intake){
+  public IntakeGamePiece(RollerIntake intake, IntakePiece type) {
     m_intake = intake; 
+    m_type = type;
     addRequirements(m_intake);
   }
 
   @Override
   public void initialize() {
-    m_intake.intake(IntakeConstants.kIntakePower);
+    if (m_type == IntakePiece.CUBE) {
+      m_intake.setIntakeMode(IntakeMode.INTAKE_CUBE);
+    } else if (m_type == IntakePiece.CONE) {
+      m_intake.setIntakeMode(IntakeMode.INTAKE_CONE);
+    }
   }
 
   @Override
   public void end(boolean interrupted) {
-    m_intake.stopIntake(); 
+    if (!interrupted) m_intake.setHeldGamePiece(m_type);
+    m_intake.setIntakeMode(IntakeMode.DISABLED);
   }
   
-  // Returns true when the command should end.
   @Override
   public boolean isFinished() {
-    return false;//m_intake.containsGamePiece(); 
+    if (m_type == IntakePiece.CUBE) {
+      return Math.abs(m_intake.getStatorCurrent()) >= IntakeConstants.kCubeIntakeCurrentStopPoint;
+    } else if (m_type == IntakePiece.CONE) {
+      return Math.abs(m_intake.getStatorCurrent()) >= IntakeConstants.kConeIntakeCurrentStopPoint;
+    }
```

Comment:
checking the currents like this probably won't work. The current spikes for an instant when first spinning up the motors, so that spike would trigger this. It probably needs to be above the current for a certain duration.

Reply:
I added a check for if some duration had passed since starting the command. Would that work?